describe/LN.ts

```diff
@@ -0,0 +1,118 @@
+import { AbstractDataAttributeDescription } from "./AbstractDataAttribute.js";
+import { LNodeType, LNodeTypeDescription } from "./LNodeType.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { describeVal, sortValues } from "./Val.js";
+
+export interface LNDescription extends NamingDescription {
+  lnType: LNodeTypeDescription;
+}
+
+/** Returns leaf data attribute (BDA or DA) from
+ * LNodeTypeDescription containing vals
+ * @param path - parent DOI/SDI/DAI name attributes
+ * @param lNodeType - LNodeTypeDescription of the logical node
+ * */
+function getLeafDataAttribute(
+  path: string[],
+  lNodeType: LNodeTypeDescription
+): Object {
+  let index = 0;
+  let dataType: any = lNodeType;
+
+  let leafData = {};
+  while (path.length > index) {
+    if (dataType.dos) {
+      // LNodeType->DO
+      const newData = dataType.dos[path[index]].type;
+      dataType = newData;
+    } else if (dataType.sdos || dataType.das) {
+      // DOType
+      if (dataType.sdos[path[index]]) {
+        //SDO
+        const sdo = dataType.sdos[path[index]];
+        const newData = sdo.type;
+        dataType = newData;
```

Comment:
Why do you use a `const newData` in each of these if/else blocks? Is it for readability? I personally would find this, for example, more readable:

```suggestion
        dataType = dataType.sdos[path[index]].type;
```

(and similarly for the other blocks which use a `const newData`)

Reply:
True, thank you. For whatever reason I thought you cannot do what you prosed.